app/models/manageiq/providers/ansible_tower/configuration_manager/configuration_script.rb

```diff
@@ -8,7 +8,7 @@ def run(vars = {})
   end
 
   def merge_extra_vars(external)
-    internal = variables.empty? ? {} : JSON.parse(variables)
+    internal = variables.empty? ? {} : variables
```

Comment:
That makes it even simpler..`{:extra_vars => variables.merge(external || {}).to_json}`